core/src/com/unciv/models/ruleset/unique/Conditionals.kt

```diff
@@ -165,6 +165,12 @@ object Conditionals {
 
             UniqueType.ConditionalBuildingBuilt ->
                 checkOnCiv { cities.any { it.cityConstructions.containsBuildingOrEquivalent(condition.params[0]) } }
+            UniqueType.ConditionalBuildingBuiltAll ->
+                checkOnCiv { cities.all { it.cityConstructions.containsBuildingOrEquivalent(condition.params[0])
```

Comment:
Noooononono
The filter for the cities *must* be applied before the .all{} otherwise irrelevant cities cause failure

Reply:
Was going to try to be funny and say
```kotlin
            checkOnCiv { cities.none { !(!it.cityConstructions.containsBuildingOrEquivalent(condition.params[0])
                    && it.matchesFilter(condition.params[1])) } }
```
is faster and lazier, but, no, looks ugly